extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/DefaultSource.scala

```diff
@@ -0,0 +1,79 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.kinesis
+
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.InitialPositionInStream
+
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.execution.datasources.CaseInsensitiveMap
+import org.apache.spark.sql.execution.streaming.Source
+import org.apache.spark.sql.sources.{DataSourceRegister, StreamSourceProvider}
+import org.apache.spark.sql.types.StructType
+
+class DefaultSource extends StreamSourceProvider with DataSourceRegister {
+
+  override def shortName(): String = "kinesis"
+
+  override def createSource(
+      sqlContext: SQLContext,
+      schema: Option[StructType],
+      providerName: String,
+      parameters: Map[String, String]): Source = {
+    val caseInsensitiveOptions = new CaseInsensitiveMap(parameters)
+    val regionName = caseInsensitiveOptions.getOrElse("regionName", {
+      throw new IllegalArgumentException("Option 'regionName' is not specified")
+    })
+    val endpointUrl = caseInsensitiveOptions.getOrElse("endpointUrl", {
+      throw new IllegalArgumentException("Option 'endpointUrl' is not specified")
+    })
+    val streamNames = caseInsensitiveOptions.getOrElse("streamNames", {
+      throw new IllegalArgumentException("Option 'streamNames' is not specified")
+    }).split(',').toSet
+    val initialPosInStream =
+      caseInsensitiveOptions.getOrElse("initialPosInStream", "LATEST") match {
+        case "LATEST" => InitialPositionInStream.LATEST
+        case "TRIM_HORIZON" => InitialPositionInStream.TRIM_HORIZON
+        case pos =>
+          throw new IllegalArgumentException(s"Unknown value of option initialPosInStream: $pos")
+      }
+
+    if ((caseInsensitiveOptions.get("AWS_ACCESS_KEY_ID").nonEmpty
+      && caseInsensitiveOptions.get("AWS_SECRET_ACCESS_KEY").isEmpty)) {
+      throw new IllegalArgumentException(
+        s"AWS_ACCESS_KEY_ID is set but AWS_SECRET_ACCESS_KEY is not found")
+    }
+    if ((caseInsensitiveOptions.get("AWS_ACCESS_KEY_ID").isEmpty
+      && caseInsensitiveOptions.get("AWS_SECRET_ACCESS_KEY").nonEmpty)) {
+      throw new IllegalArgumentException(
+        s"AWS_SECRET_ACCESS_KEY is set but AWS_ACCESS_KEY_ID is not found")
+    }
```

Comment:
Check whether access keys are available (either specified, or loaded from a provider) right here. Otherwise, it fails later, and gives a bad error message like 
```
com.amazonaws.AmazonClientException: Unable to load AWS credentials from any provider in the chain
	at com.amazonaws.auth.AWSCredentialsProviderChain.getCredentials(AWSCredentialsProviderChain.java:117)
	at org.apache.spark.streaming.kinesis.KinesisSource$$anonfun$5.apply(KinesisSource.scala:85)
```
So better to resolve the credentials right here and pass it on to the source without an option.